agents/xy_stage/xy_ocs_agent.py

```diff
@@ -0,0 +1,243 @@
+                    if not acquired:
+                        self.log.warn(f"Could not check because lock held by {self.lock.job}")
+                        return False, "Could not acquire lock"
+                    self.is_moving = self.xy_stage.moving
+            
+            if not self.is_moving:
+                break
+        return True, "X Move Complete"
+
+    def move_y_cm(self, session, params):
+        """
+        params: dict: { 'distance': float, 'velocity':float < 1.2}
+        """
+
+        with self.lock.acquire_timeout(timeout=3, job='move_y_cm') as acquired:
+            if not acquired:
+                self.log.warn(f"Could not start y move because lock held by {self.lock.job}")
+                return False, "could not acquire lock"
+            self.xy_stage.move_y_cm( params['distance'], params['velocity'])
+        
+        time.sleep(1)
+        while True:
+            ## data acquisition updates the moving field if it is running
+            if not self.take_data:
+                with self.lock.acquire_timeout(timeout=3, job='move_y_cm') as acquired:
+                    if not acquired:
+                        self.log.warn(f"Could not check for move because lock held by {self.lock.job}")
+                        return False, "could not acquire lock"
+                    self.is_moving = self.xy_stage.moving
+            if not self.is_moving:
+                break
+        return True, "Y Move Complete"
+
+ 
+    def set_position(self, session, params):
+        """
+        params: dict: {'position': (float, float)}
+        """
+        with self.lock.acquire_timeout(timeout=3, job='set_position') as acquired:
+            if not acquired:
+                self.log.warn(f"Could not set position because lock held by {self.lock.job}")
+                return False, "Could not acquire lock"
+                        
+            self.xy_stage.position = params['position']
+        return True, "Position Updated"
+
+    def start_acq(self, session, params=None):
+        """
+        params: dict: {`sampling_frequency': float, sampling rate in Hz}
+        """
+        if params is None:
+            params = {}
+
+        f_sample = params.get('sampling_frequency', self.sampling_frequency)
+        sleep_time = 1/f_sample - 0.1
+        if not self.initialized:
+            self.init_xy_stage_task(session)
+        
+        with self.lock.acquire_timeout(timeout=0, job='acq') as acquired:
+            if not acquired:
+                self.log.warn("Could not start acq because {} is already running".format(self.lock.job))
+                return False, "Could not acquire lock."
+
+            self.log.info(f"Starting Data Acquisition for XY Stages at {f_sample} Hz")
+            session.set_status('running')
+            self.take_data = True
+            last_release = time.time()
+
+            while self.take_data:
+                if time.time()-last_release > 1.:
```

Comment:
I believe you need to update `last_release` inside this block...

Reply:
oh truth. shall fix

Reply:
Thanks for the tip!

Reply:
Great.  Last thing (before Brian takes over :) ) -- that sleep(random.uniform(...)) in the Pacemaker docs is actually supposed to be simulating your instrument's random sampling time... you don't need it in your agent (you have a real device to provide the variance!).  Maybe @jlashner wants to clarify the docs on that :)

Reply:
Alright, gotcha. Thanks!